install router as part of installing openshift 3

pweil- opened the pull request:
Install router as part of doing an install-openshift3.

Changes:

1.  Ensure the install command does a build and starts the service
2.  Add an action that will check for openshift in a running state and run the install command for a default router.  Action should skip installation if default router already exists.
3.  Change try-restart to restart.  Try restart does nothing if the service is not already running.

pweil- commented:
@danmcp - as requested.

danmcp commented on lib/vagrant-openshift/action.rb:
```diff
@@ -37,6 +37,9 @@ def self.install_openshift3(options)
           b.use InstallOpenshift3
           b.use InstallOpenshift3Rhel7
           b.use InstallOpenshift3AssetDependencies
+          b.use BuildOpenshift3, options
```
build is currently being called as a separate step by all the scripts.

danmcp commented on lib/vagrant-openshift/action/try_restart_openshift3.rb:
```diff
@@ -26,7 +26,7 @@ def initialize(app, env)
 
         def call(env)
-          sudo(env[:machine], "systemctl try-restart openshift")
+          sudo(env[:machine], "systemctl restart openshift")
```
We don't want to start openshift unless it's already running with this command.

pweil- replied:
So my issue, then, is that in order to install a router I need openshift to be running.  Since we were shooting for the all-in-one experience I forced it to come up rather than ignore if it wasn't started.  Should I just let the router install fail if it is not up?

danmcp replied:
@pweil- I am thinking you want a separate action that starts openshift, installs the router, and stops openshift.  And then add that step to jenkins.  That sound reasonable?

pweil- replied:
Sounds good to me.  Thanks!

pweil- commented:
Added a `vagrant install-openshift3-router` command that will start openshift, install the router, and stop openshift.  Reverted the try-restart command to not force the server to come up.

pweil- commented:
@danmcp - PTAL.  I'll talk to @ironcladlou about updating the Jenkins environment to use this although we may want to consider having it not use it only so it can run the route install command manually in the e2e test (as it does now)

danmcp commented:
@pweil- I should have mentioned this before, but rather than adding two new commands for stop and start.  Let's add 1 new command that takes stop/start/restart/whatever as an arg.  Then switch over the existing calls to try-restart to use that and use it from all the new locations.

pweil- commented:
Updated to use a generic RunSystemctl command and pass action/service.  Refactored all try-restarts to use RunSystemctl
